[SVG] Use `currentViewBoxRect()` for intrinsic ratio calculation

Ahmad-S792 opened the pull request:
#### 372b21fd954177d09e408954276cf1201cdf9da7
<pre>
[SVG] Use `currentViewBoxRect()` for intrinsic ratio calculation
<a href="https://bugs.webkit.org/show_bug.cgi?id=296548">https://bugs.webkit.org/show_bug.cgi?id=296548</a>
<a href="https://rdar.apple.com/problem/157445966">rdar://problem/157445966</a>

Reviewed by Brent Fulgham.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch updates computeIntrinsicSizeAndPreferredAspectRatio() to use
currentViewBoxRect() when computing the intrinsic aspect ratio. This
ensures that an &lt;svg&gt; element referencing a &lt;view&gt; (via fragment
identifier) correctly derives its aspect ratio from the viewBox defined
in the &lt;view&gt; element.

* Source/WebCore/rendering/svg/RenderSVGRoot.cpp:
(WebCore::RenderSVGRoot::computeIntrinsicSizeAndPreferredAspectRatio const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.cpp:
(WebCore::LegacyRenderSVGRoot::computeIntrinsicSizeAndPreferredAspectRatio const):
* LayoutTests/TestExpectations: Progression on both LegacySVG and LBSE

Canonical link: <a href="https://commits.webkit.org/298307@main">https://commits.webkit.org/298307@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9da924ea460bbb878d3b417ffd963a7bc6af1984

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115078 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34791 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25275 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121193 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65718 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b7473c08-0368-439a-904f-67f917658ec2) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116967 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35438 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43353 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87448 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42247 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e7429706-5d4b-42cc-8f14-78924d145898) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118026 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28247 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103319 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67859 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27418 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21439 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64846 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97631 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21556 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124381 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42047 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31445 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96245 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42417 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99509 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96031 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24436 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41233 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19077 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/38046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41922 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47458 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41463 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44782 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43198 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.cpp:
```diff
@@ -108,7 +108,7 @@ std::pair<FloatSize, FloatSize> LegacyRenderSVGRoot::computeIntrinsicSizeAndPref
         preferredAspectRatio = { intrinsicSize.width(), intrinsicSize.height() };
     else {
-        FloatSize viewBoxSize = svgSVGElement().viewBox().size();
+        FloatSize viewBoxSize = svgSVGElement().currentViewBoxRect().size();
```
Is this change needed in `rendering/svg/RenderSVGRoot.cpp`, too?

brentfulgham replied:
This change brings our code in alignment with Blink's `LayoutSVGRoot::UnscaledNaturalSizingInfo(...)`, so seems correct.

Ahmad-S792 replied:
On it - to check for LBSE.

Ahmad-S792 replied:
@brentfulgham - Done - confirmed by locally compiling build with LBSE on that it progresses LBSE as well.

brentfulgham reviewed (changes requested):
I think this looks correct, but I believe you need to make the same change in `rendering/svg/RenderSVGRoot.cpp` as well. Can you please add that change to this patch to ensure we get the same benefit when LBSE is turned on?

brentfulgham reviewed (approved):
Thank you for fixing LBSE as well! r=me

Ahmad-S792 committed:
[SVG] Use `currentViewBoxRect()` for intrinsic ratio calculation
https://bugs.webkit.org/show_bug.cgi?id=296548
rdar://problem/157445966

Reviewed by Brent Fulgham.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch updates computeIntrinsicSizeAndPreferredAspectRatio() to use
currentViewBoxRect() when computing the intrinsic aspect ratio. This
ensures that an <svg> element referencing a <view> (via fragment
identifier) correctly derives its aspect ratio from the viewBox defined
in the <view> element.

* Source/WebCore/rendering/svg/RenderSVGRoot.cpp:
(WebCore::RenderSVGRoot::computeIntrinsicSizeAndPreferredAspectRatio const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.cpp:
(WebCore::LegacyRenderSVGRoot::computeIntrinsicSizeAndPreferredAspectRatio const):
* LayoutTests/TestExpectations: Progression on both LegacySVG and LBSE

Canonical link: https://commits.webkit.org/298307@main
